### PR TITLE
Remove EventTimeVector... and more!

### PR DESCRIPTION
**Pull request by ararslan**
Basically a laundry list of stuff I've been meaning to do.

* Nix `EventTimeVector`—it's annoying and unnecessary
* Drop -pre from the Julia lower bound
* `EventTime(x)` should be an observed event

That last point was a really bad choice initially, not sure why I ever thought the current behavior was a good idea.

I also worked around the 0.7 test changes.

### CODE REVIEW

File: test/runtests.jl [hunk 1]
```diff
@@ -1,85 +1,25 @@
 using Survival
-using Base.Test
+if VERSION >= v"0.7.0-DEV.2004"
```
Comment (nalimilan) [inline test/runtests.jl]: I recommended using `Compat.Test` in Distributions, so... :-)

File: test/runtests.jl [hunk 2]
```diff
@@ -1,85 +1,25 @@
...
-
-            sort!(x)
-            @test x == EventTimeVector(collect(1:6))
-
-            y = EventTimeVector([7])
-            @test [x; y] == EventTimeVector(collect(1:7))
-            @test [y; y; y] == EventTimeVector([7, 7, 7])
-        end
-    end
+    @test convert(Int, EventTime(1)) == 1
```
Comment (nalimilan) [inline test/runtests.jl]: Maybe also test with a censored event? Also with a target type different from the input eltype.

File: src/eventtimes.jl [hunk 3]
```diff
@@ -6,14 +6,14 @@
     EventTime{T}
 
 Immutable object containing the real-valued time to an event as well as an indicator of
-whether the time corresponds to an observed event or right censoring.
+whether the time corresponds to an observed event (`true`) or right censoring (`false`).
 """
 struct EventTime{T<:Real}
```
Comment (nalimilan) [inline src/eventtimes.jl]: I can't help wondering whether this wouldn't be another case where using `Union{Event{T}, Censored{T}}` wouldn't make sense, with one type for uncensored and one type for censored events. That would probably be more confusing that helpful, but I thought I would mention it. At least the advantage would be that the memory layout of arrays of event times would be efficient (no additional padding to align e.g. 64 + 8 bits).

Comment (ararslan) [reply src/eventtimes.jl]: Hm, that's an interesting idea. Having a separate type for right censoring might be a good distinction from left truncation or whatever else should we ever implement those, but I also don't want to be in another `Nullable{T}` situation, where everything requires unwrapping to be useful. (Though I guess `EventTime` is like that now...)

File: test/runtests.jl (hunk 2 above)
Comment (ararslan) [reply test/runtests.jl]: Good call!

File: test/runtests.jl (hunk 1 above)
Comment (ararslan) [reply test/runtests.jl]: HAH yeah. That's indeed a better idea.

**Comment by ararslan** [comment]
I just realized that I removed `EventTimeVector` with the implicit replacement of `Vector{EventTime{T}}` but I didn't actually define any methods for things in terms of `Vector{EventTime{T}}`...

**Comment by ararslan** [comment]
Finally got around to finishing this up. The CI failures on 0.7 are unrelated.

**Pull request merged**